js_modules/dagster-ui/packages/ui-core/src/assets/AssetEvents.tsx

```diff
@@ -199,3 +229,71 @@ export const AssetEvents: React.FC<Props> = ({
     </>
   );
 };
+
+type EventType = 'observation' | 'materialization';
+
+const ALL_EVENT_TYPES: EventType[] = ['observation', 'materialization'];
+
+export const EventTypeSelect: React.FC<{
+  value: EventType[];
+  onChange: (value: EventType[]) => void;
+}> = ({value, onChange}) => {
+  const [showMenu, setShowMenu] = React.useState(false);
+
+  const onToggle = (type: EventType) => {
+    if (value.includes(type)) {
+      onChange(value.filter((v) => v !== type));
+    } else {
+      onChange([...value, type]);
+    }
+  };
+
+  return (
+    <Popover
+      isOpen={showMenu}
+      placement="bottom-start"
+      canEscapeKeyClose
+      onInteraction={(nextOpenState: boolean) => setShowMenu(nextOpenState)}
+      content={
+        <Menu style={{width: 140}} aria-label="filter-options">
+          <MenuItem
+            shouldDismissPopover={false}
+            onClick={() => onToggle('materialization')}
+            text={
+              <Box padding={{horizontal: 2}} flex={{direction: 'row', alignItems: 'center'}}>
+                <Checkbox
+                  size="small"
+                  checked={value.includes('materialization')}
+                  onChange={() => {}}
+                  label="Materialization"
+                />
+              </Box>
+            }
+          />
+          <MenuItem
+            shouldDismissPopover={false}
+            onClick={() => onToggle('observation')}
+            text={
+              <Box padding={{horizontal: 2}} flex={{direction: 'row', alignItems: 'center'}}>
+                <Checkbox
+                  size="small"
+                  checked={value.includes('observation')}
+                  onChange={() => {}}
+                  label="Observation"
+                />
+              </Box>
+            }
+          />
+        </Menu>
+      }
+    >
+      <Button
+        onClick={() => setShowMenu((current) => !current)}
+        icon={<Icon name="filter_alt" />}
+        rightIcon={<Icon name="expand_more" />}
+      >
+        Type ({value.length})
+      </Button>
+    </Popover>
+  );
+};
```

Comment:
Will there ever be more than two of these? Could it just be two checkboxes side-by-side instead of the popover+menu?

Reply:
Yeah I had a version like this as well actually - @braunjj preferred the dropdown for consistency but let me ask him again real quick

![image](https://github.com/dagster-io/dagster/assets/1037212/3c34c1b9-494f-4112-bbd0-76484a4b9699)

Reply:
Yeah. I personally prefer the dropdown version because it's cleaner. We may be adding a new event type for `data quality checks` here so this list may end up expanding.